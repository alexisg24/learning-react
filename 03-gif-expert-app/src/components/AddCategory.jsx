import { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('')
  const handleInputChange = (event) => {
    setInputValue(event.target.value)
  }
  const handleSubmit = (event) => {
    event.preventDefault()
    if (inputValue.trim().length <= 1) return true
    onNewCategory(inputValue.trim())
    setInputValue('')
  }

  return (
    <form onSubmit={handleSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Search Gifs'
        value={inputValue}
        onChange={handleInputChange}
      />
      <button>Add</button>
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}

export { AddCategory }
