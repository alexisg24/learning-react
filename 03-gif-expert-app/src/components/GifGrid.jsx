import { GifItem } from './'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types'

const GifGrid = ({ category }) => {
  const { images = [], isLoading } = useFetchGifs(category)
  return (
    <>
      {isLoading && <h2 style={{ textAlign: 'center' }}>Loading...</h2>}
      <p className='category'>{category}</p>
      <section className='card-grid'>
        {
          images.map((img) => (
            <GifItem
              key={img.id}
              {...img}
            />
          ))
        }
      </section>
    </>
  )
}
GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}

export { GifGrid }
