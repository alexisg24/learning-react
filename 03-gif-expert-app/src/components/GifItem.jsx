import PropTypes from 'prop-types'
export const GifItem = ({ title, url }) => {
  return (
    <article className='card'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </article>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
