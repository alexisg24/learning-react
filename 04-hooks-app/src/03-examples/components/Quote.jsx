import PropTypes from 'prop-types'
import { useLayoutEffect, useRef, useState } from 'react'
const Quote = ({ quote = 'Texto', author = 'Alexis G' }) => {
  const pRef = useRef()
  const [boxSize, setBoxSize] = useState({ width: 0, height: 0 })
  useLayoutEffect(() => {
    const { width, height } = pRef.current.getBoundingClientRect()
    setBoxSize({ width, height })
  }, [quote])

  return (
    <>
      <blockquote
        className='blockquote text-end'
        style={{ display: 'flex' }}
      >
        <p
          ref={pRef}
          className='mb-1'
        >{quote}
        </p>
        <footer className='blockquote-footer'>{author}</footer>
      </blockquote>
      <code className='mr-2'>{JSON.stringify(boxSize)}</code>
    </>
  )
}

Quote.propTypes = {
  author: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired
}

export { Quote }
