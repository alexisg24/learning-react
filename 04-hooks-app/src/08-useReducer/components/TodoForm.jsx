import { useForm } from '../../hooks/useForm'

const TodoForm = ({ handleSubmit }) => {
  const { description, onInputChange, onResetForm } = useForm({
    description: ''
  })

  const submitEvent = (event) => {
    event.preventDefault()
    if (description.length <= 1) return
    const newTodo = {
      id: new Date().getTime(),
      description,
      done: false
    }
    handleSubmit(newTodo)
    onResetForm()
  }
  return (
    <>
      <form
        onSubmit={submitEvent}
      >
        <input
          type='text'
          placeholder='Add TODO'
          className='form-control'
          name='description'
          onChange={onInputChange}
          value={description}
        />

        <button
          type='submit'
          className='btn btn-outline-primary mt-2'
        >Add TODO
        </button>
      </form>
    </>
  )
}

export { TodoForm }
