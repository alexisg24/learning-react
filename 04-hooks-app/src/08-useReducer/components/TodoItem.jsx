const TodoItem = ({ id, description, done, handleDelete, handleToggle }) => {
  return (
    <>
      <li className='list-group-item d-flex justify-content-between align-items-center'>
        <span
          className={`${(done) ? 'text-decoration-line-through' : ''}`}
          onClick={() => handleToggle(id)}
        >{description}
        </span>
        <button
          className='btn btn-outline-danger'
          onClick={() => handleDelete(id)}
        >Delete
        </button>
      </li>
    </>
  )
}

export { TodoItem }
