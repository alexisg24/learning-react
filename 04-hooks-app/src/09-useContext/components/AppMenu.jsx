import { Link, NavLink } from 'react-router-dom'

const AppMenu = () => {
  const handleActive = ({ isActive }) => `nav-link ${isActive ? 'active' : ''}`
  return (
    <>
      <nav className='navbar navbar-expand-lg bg-dark navbar-dark rounded-3'>
        <div className='container-fluid'>
          <Link className='navbar-brand' to='/'>ContextApp</Link>
          <div className='collapse navbar-collapse' id='navbarNav'>
            <ul className='navbar-nav'>

              <NavLink
                to='/'
                className={handleActive}
              >
                Home
              </NavLink>

              <NavLink
                to='/about'
                className={handleActive}
              >
                About
              </NavLink>

              <NavLink
                to='/login'
                className={handleActive}
              >
                Login
              </NavLink>

            </ul>
          </div>
        </div>
      </nav>
    </>
  )
}

export { AppMenu }
