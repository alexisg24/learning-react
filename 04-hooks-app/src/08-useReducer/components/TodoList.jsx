import { TodoItem } from './TodoItem'

const TodoList = ({ todos = [], handleDelete, handleToggle }) => {
  return (
    <>
      <ul className='list-group'>
        {
            todos.map((todo) => (<TodoItem key={todo.id} {...todo} handleDelete={handleDelete} handleToggle={handleToggle} />))
        }
      </ul>
    </>
  )
}

export { TodoList }
