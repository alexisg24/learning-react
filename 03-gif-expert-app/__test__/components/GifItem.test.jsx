import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'
const props = {
  title: 'My Title',
  url: 'https://media1.giphy.com/media/hWdPJwIDnKGvSj6dAC/giphy.gif?cid=ecf05e47b167d0f998d7eb99ddc37eb339a2b96f811be067&ep=v1_gifs_gifId&rid=giphy.gif&ct=s'
}
describe('Tests in GifItem.jsx', () => {
  test('should match with the snapshot', () => {
    const { container } = render(<GifItem {...props} />)
    expect(container).toMatchSnapshot()
  })

  test('should show the img with url and alt', () => {
    render(<GifItem {...props} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(props.url)
    expect(alt).toBe(props.title)
  })

  test('Title need to be a text', () => {
    render(<GifItem {...props} />)
    expect(screen.getByText(props.title)).toBeTruthy()
  })
})
