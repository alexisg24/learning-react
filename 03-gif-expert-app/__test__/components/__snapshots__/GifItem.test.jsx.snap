// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifItem.jsx should match with the snapshot 1`] = `
<div>
  <article
    class="card"
  >
    <img
      alt="My Title"
      src="https://media1.giphy.com/media/hWdPJwIDnKGvSj6dAC/giphy.gif?cid=ecf05e47b167d0f998d7eb99ddc37eb339a2b96f811be067&ep=v1_gifs_gifId&rid=giphy.gif&ct=s"
    />
    <p>
      My Title
    </p>
  </article>
</div>
`;
